### ISSUE TITLE
Support for OpenAPI Specification - 'format' property for 'password' and 'email'

### ISSUE THREAD

**Issue by DeepikaHadadi** [specmatic/specmatic#946]
**Description**
While using an OpenAPI Specification YAML file, the 'format' field is not supported for types such as 'password' and 'email'. 

According to the OpenAPI Specification (OAS), the 'format' field is open and can be used to validate input or map values to specific types in a programming language. However, it appears that Specmatic does not support specific formats like 'email' and 'password', ignoring the 'format'. 

**Steps to reproduce**
### Scenario 1: With Format field have value as 'email' and 'password'

<details>
  <summary><b>Example Specification: With format field as 'email' and 'password'</b></summary>

```yaml
servers:
  - url: http://localhost:3000
paths:
  /api/createuser:
    post:
      tags:
        - User Creation
      summary: Create a new user.
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/User"
      responses:
        "201":
          description: User account successfully created.
          content:
            application/json:
              schema:
                type: object
                properties:
                  message:
                    type: string
                    example: "User created successfully."
                  userId:
                    type: number
                    example: 123
 
components:
  schemas:
    User:
      type: object
      required:
        - username
        - password
        - email
      properties:
        username:
          type: string
          description: Desired username for the account.
          minLength: 5
          maxLength: 25
          example: joewilson
          pattern: "^[a-z]*$"
        password:
          type: string
          format: password
          description: Password for the account.
          minLength: 8
          maxLength: 25
          example: Password01
        email:
          type: string
          format: email
          description: Email address for account verification and communication.
          maxLength: 255
          example: joe@testdomain.com
          pattern: "^[a-z]*$@gmail.com"

``` 
</details>

**Command to run the test**: `npx specmatic test createUser.yml --testBaseURL=
http://localhost:3000`

<details>
  <summary><b> Response </b></summary>

```
Request to http://localhost:3000 at 2024-1-31 1:12:32.507
    POST /api/createuser
    Accept-Charset: UTF-8
    Accept: */*
    Content-Type: application/json
    
    {
        "email": {
        },
        "password": {
        },
        "username": "RYOOF"
    }


  Response at 2024-1-31 1:12:32.561
    500 Internal Server Error
    X-Powered-By: Express
    Content-Security-Policy: default-src 'none'
    X-Content-Type-Options: nosniff
    Content-Type: text/html; charset=utf-8
    Content-Length: 1725
    Set-Cookie: connect.sid=s%3ACKQyQz_BtM2FUonhlygtjEsTHY_a_Zmm.0tQ3vrJtdFwWE7AFK6Z3UpMNP6HgH6s2bAvZW%2BxyMAE; Path=/; HttpOnly
    Date: Wed, 31 Jan 2024 02:12:32 GMT
    Connection: keep-alive
    Keep-Alive: timeout=5
```
</details>

### Scenario 2: With Format field as `string`

<details>
  <summary><b>Example Specification: With format field as string</b></summary>

```yaml

servers:
  - url: http://localhost:3000
paths:
  /api/createuser:
    post:
      tags:
        - User Creation
      summary: Create a new user.
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/User"
      responses:
        "201":
          description: User account successfully created.
          content:
            application/json:
              schema:
                type: object
                properties:
                  message:
                    type: string
                    example: "User created successfully."
                  userId:
                    type: number
                    example: 123
 
components:
  schemas:
    User:
      type: object
      required:
        - username
        - password
        - email
      properties:
        username:
          type: string
          description: Desired username for the account.
          minLength: 5
          maxLength: 25
          example: joewilson
          pattern: "^[a-z]*$"
        password:
          type: string
          format: string
          description: Password for the account.
          minLength: 8
          maxLength: 25
          example: Password01
        email:
          type: string
          format: string
          description: Email address for account verification and communication.
          maxLength: 255
          example: joe@testdomain.com
          pattern: "^[a-z]*$@gmail.com"

``` 
</details>

**Command to run the test**: `npx specmatic test createUser.yml --testBaseURL=
http://localhost:3000`

<details>
  <summary><b> Response </b></summary>

```
Request to http://localhost:3000 at 2024-1-31 1:14:3.610
    POST /api/createuser
    Accept-Charset: UTF-8
    Accept: */*
    Content-Type: application/json
    
    {
        "email": "OQXIV",
        "password": "HCYKWMPT",
        "username": "UQTQU"
    }


  Response at 2024-1-31 1:14:3.619
    201 Created
    X-Powered-By: Express
    Content-Type: application/json; charset=utf-8
    Content-Length: 56
    ETag: W/"38-GU8Q9mNu2zG5mhH8YsREsY1kHM4"
    Set-Cookie: connect.sid=s%3Al9XTqv9GjXp0KofMJrOGlIdwtkgZ9VSK.rm%2BIACiSOafPXDGGAvp3QE2vJuCck8S8LVQq58OPMx0; Path=/; HttpOnly
    Date: Wed, 31 Jan 2024 02:14:03 GMT
    Connection: keep-alive
    Keep-Alive: timeout=5
    
    {
        "message": "User UQTQU added successfully!",
        "userId": 42
    }
```
</details>

**Expected behavior**
Specmatic should support format of type 'email' , 'password'

**Screenshots**
Scenario 1 Example screeenshot
<img width="833" alt="image" src="https://github.com/znsio/specmatic/assets/52940752/cb6bafe1-8edb-460e-9877-642398e69c2e">
Scenario 2 Example screenshot
<img width="818" alt="image" src="https://github.com/znsio/specmatic/assets/52940752/fe1739e9-94d4-4551-8867-cb8950d009d4">

**System Information:**
 - OS & version: MacOS 
 - Browser & version: [e.g. chrome, safari]
 - Specmatic version: 1.2.12
 - JDK version:  17

**Additional context**

**Comment by harikrishnan83** [comment]
hello @DeepikaHadadi, thanks a lot for all the details shared in the bug. We will certainly prioritise this.

**Comment by harikrishnan83** [comment]
@DeepikaHadadi The fix is now available as part of [release 1.2.15](https://github.com/znsio/specmatic/releases/tag/1.2.15).
